### DISCUSSION TITLE
Support new React Native Architecture?

### DISCUSSION THREAD

**Discussion by fabiendem** [innoveit/react-native-ble-manager#902] [Ideas]
Hello,

I am wondering if there are plans to support the new RN architecture in the future.
https://reactnative.dev/docs/new-architecture-intro

This library `react-native-ble-manager` is now de-facto the best option for BLE support on React Native.
There is another BLE library out there (https://github.com/dotintent/react-native-ble-plx) but imo it looks like it's no longer maintained.

Let us know how we could help.

Cheers

**Comment by marcosinigaglia** [comment]
Hi @fabiendem we have no plans to do this anytime soon, any PR is welcome.

**Reply by fabiendem** [reply]
OK, thanks for your prompt reply.
This is not on our roadmap, but we will be in touch if it happens.

**Reply by fabiendem** [reply]
FYI, downloads trend comparison: https://npmtrends.com/react-native-ble-manager-vs-react-native-ble-plx

**Comment by AshlandWest** [comment]
Hello,
Is this discussion still relevant and am I reading this correctly to interpret that this library will not work with the latest version of React Native?
I am using React Native 0.70.5 and having trouble writing data to the device. I can connect to peripherals, but I am now wondering if my issue is due to an incompatible RN version.

Kind regards,
Ash

**Reply by AshlandWest** [reply]
UPDATE: The write method works fine on 0.70.5. I resolved my issue. I would still like to know if I can expect any pitfalls with the new version.

Kind regards,
Ash
